File: files/problems/js/student-attendance-record.js
// by @codeAbinash
// Time Complexity : O(n)
// Space Complexity : O(1)

const checkRecord = s => {
    let countAbsent = 0
    for (let ch of s) if (ch === 'A') countAbsent++;
    return !s.includes('LLL') && countAbsent < 2;
};

File: files/problems/cpp/student-attendance-record.cpp
// by @codeAbinash
// Time Complexity : O(n)
// Space Complexity : O(1)

class Solution {
   public:
    bool checkRecord(string s) {
        int absent = 0;
        int late = 0;
        for (auto ch : s) {
            if (late >= 3) return 0;
            if (ch == 'A') {
                absent++;
                late = 0;
            } else if (ch == 'L')
                late++;
            else
                late = 0;
        }
        return absent < 2 && late < 3;
    }
};